### PR TITLE
Export Source classes declarations.

### PR DESCRIPTION
**Pull request by sebisteiner**
In MapLibre v1 the various Source types were exported. See here: https://github.com/maplibre/maplibre-gl-js/blob/v1.15.2/src/index.d.ts#L1234

I'd like to restore this behavior so that the types can be used directly in applications using MapLibre.

Also see discussion here: https://github.com/maplibre/maplibre-gl-js/issues/801

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Manually test the debug page.
 - [ ] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [ ] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -18,13 +18,20 @@ import {Evented} from './util/evented';
 import config from './util/config';
 import {Debug} from './util/debug';
 import {isSafari} from './util/util';
-import {setRTLTextPlugin, getRTLTextPluginStatus} from './source/rtl_text_plugin';
+import {getRTLTextPluginStatus, setRTLTextPlugin} from './source/rtl_text_plugin';
```
Comment (HarelM) [inline src/index.ts:21]: Can this be revered? I don't see actual change...

Comment (sebisteiner) [reply src/index.ts]: I reverted the import order.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ### Features and improvements
 
-- *...Add new stuff here...*
+- Export `Source` classes (`GeoJSONSource` etc.) declarations.
```
Comment (HarelM) [inline CHANGELOG.md:5]: Thanks! please link to the relevant issue (I guess it is #801, right?)

**Comment by HarelM** [comment]
Please improve the changelog message to have the issue number inside "()" and I'll merge.
Thanks for your time!!

**Review by HarelM — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (sebisteiner) [reply CHANGELOG.md]: Done. Thanks for your support.

**Pull request merged**